pkg/deploy/strategy/support/lifecycle.go

```diff
@@ -82,9 +123,11 @@ func (e *HookExecutor) Execute(hook *deployapi.LifecycleHook, deployment *kapi.R
 	// Retry failures are treated the same as Abort.
 	switch hook.FailurePolicy {
 	case deployapi.LifecycleHookFailurePolicyAbort, deployapi.LifecycleHookFailurePolicyRetry:
-		return fmt.Errorf("%s hook failed, aborting deployment: %s", label, err)
+		e.event(deployment, kapi.EventTypeWarning, "Failed", fmt.Sprintf("The %s hook failed: %v, aborting deployment %s/%s", label, err, deployment.Namespace, deployment.Name))
```

Comment:
Yes the hook. OK I will move the events to DC then and make sure the RC
name is part of the event message.

On Tue, Jun 28, 2016 at 12:02 PM, Michail Kargakis <notifications@github.com
> wrote:

> In pkg/deploy/strategy/support/lifecycle.go
> <https://github.com/openshift/origin/pull/9456#discussion_r68729882>:
>
> > @@ -82,9 +123,11 @@ func (e *HookExecutor) Execute(hook *deployapi.LifecycleHook, deployment *kapi.R
> >  	// Retry failures are treated the same as Abort.
> >  	switch hook.FailurePolicy {
> >  	case deployapi.LifecycleHookFailurePolicyAbort, deployapi.LifecycleHookFailurePolicyRetry:
> > -		return fmt.Errorf("%s hook failed, aborting deployment: %s", label, err)
> > +		e.event(deployment, kapi.EventTypeWarning, "Failed", fmt.Sprintf("The %s hook failed: %v, aborting deployment %s/%s", label, err, deployment.Namespace, deployment.Name))
>
> My main concern with having these events as part of the replication
> controller is that then we will end up with fragmented information, because
> we already moved all of the other events regarding the deployment process
> on the config: #7830 <https://github.com/openshift/origin/pull/7830>
>
> It is also consistent with what's happening upstream:
>
> https://github.com/kubernetes/kubernetes/blob/b6f966f8f5b4903342fa0aec6f63b1fec3c030ea/pkg/controller/deployment/deployment_controller.go#L1368
>
> @mfojtik <https://github.com/mfojtik> by pod creation you mean the hook?
> We should move this too on the config if that's not the case currently.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9456/files/a9e5f081265c3c5460da1eddee8d16ec7da674ec#r68729882>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AACsaBUkDWCAFOu10NpokCJRAViw0Qqpks5qQPFSgaJpZM4I6lMS>
> .
>



-- 
-----------------------------------------------
Michal Fojtik <mi@mifo.sk <mfojtik@mifo.sk>>
Red Hat OpenShift, Engineering